File: src/NubipApp.Models/Domain/Group.cs
using System.Collections.Generic;

namespace NubipApp.Models.Domain
{
    public class Group
    {
        public int GroupId { get; set; }
        public string Title { get; set; }
        public int SpecialityId { get; set; }
        public int TeacherId { get; set; }
        public int EducationLevelId { get; set; }
        public int EducationFormId { get; set; }
        public int StudentCount { get; set; }
        public bool IsActive { get; set; }

        public Speciality Speciality { get; set; }
        public Teacher Teacher { get; set; }
        public EducationLevel EducationLevel { get; set; }
        public EducationForm EducationForm { get; set; }
        
        public ICollection<Lesson> Lessons { get; set; }
    }
}

File: src/models/group.ts
import mongoose from "mongoose";

let Schema = mongoose.Schema;

let GroupSchema = new Schema({
	groupId: Number,
	title: String,
	specialityId: [{
		type: Schema.Types.ObjectId, ref: 'Speciality'
	}],
	teacherId: [{
		type: Schema.Types.ObjectId, ref: 'Teacher'
	}],
	educationLevelId: [{
		type: Schema.Types.ObjectId, ref: 'EducationLevel'
	}],
	educationFormId: [{
		type: Schema.Types.ObjectId, ref: 'EducationForm'
	}],
	studentCount: Number,
	isActive: Boolean,
});

let Group = mongoose.model("Group", GroupSchema);

export default Group;